lambdas/csv_to_glue_catalog/main.py

```diff
@@ -0,0 +1,340 @@
+    """Deduplicate column names by appending a counter to duplicate names.
+
+    Args:
+        columns: List of column names (may contain duplicates)
+
+    Returns:
+        List of deduplicated column names
+    """
+    deduped_headers = []
+    header_counts = {}
+
+    for col in columns:
+        if col in header_counts:
+            header_counts[col] += 1
+            deduped_headers.append(f"{col}_{header_counts[col]}")
+        else:
+            header_counts[col] = 0
+            deduped_headers.append(col)
+
+    return deduped_headers
+
+
+def extract_csv_column_definitions(bucket: str, key: str) -> dict[str, str]:
+    """Extract column names from CSV header, normalize and deduplicate them.
+
+    Returns dictionary mapping column names to types (all string type).
+
+    Args:
+        bucket: S3 bucket name
+        key: S3 object key (file path)
+
+    Returns:
+        Dictionary mapping column names to types (all 'string')
+
+    Raises:
+        ValueError: If CSV file has no header row or empty header row
+    """
+    s3_path = f"s3://{bucket}/{key}"
+
+    try:
+        df = wr.s3.read_csv(
+            path=s3_path,
+            nrows=1,
+            use_threads=False,
+            encoding="utf-8-sig",
+            on_bad_lines="skip",
+        )
+    except Exception as e:
+        logger.error(f"Failed to read CSV from S3: {e}")
+        raise ValueError(f"Unable to read CSV file: {e}") from e
+
+    if len(df.columns) == 0:
+        raise ValueError("CSV file has no header row")
+
+    column_names = list(df.columns)
+
+    if not column_names or all(not col or col.strip() == "" for col in column_names):
+        raise ValueError("CSV file has empty or invalid header row")
+
+    normalized_headers = [normalize_name(col) for col in column_names]
+    deduped_headers = deduplicate_column_names(normalized_headers)
+
+    columns_types = {}
+    for col_name in deduped_headers:
+        if not col_name:
+            col_name = f"column_{len(columns_types)}"
+        columns_types[col_name] = "string"
+
+    logger.info(
+        f"A total of {len(columns_types)} column definitions were extracted from CSV header"
+    )
+    return columns_types
+
+
+def create_glue_table(
+    database_name: str,
+    table_name: str,
+    bucket: str,
+    s3_key: str,
+    columns_types: dict[str, str],
+):
+    """Create or recreate Glue Catalog table using AWS Data Wrangler.
+
+    Args:
+        database_name: Glue database name
+        table_name: Glue table name
+        bucket: S3 bucket name
+        s3_key: S3 object key (file path)
+        columns_types: Dictionary mapping column names to types
+    """
+    # Extract directory path: s3://bucket/parking/user/file.csv -> s3://bucket/parking/user/
+    s3_path_parts = s3_key.rsplit("/", 1)
+    s3_directory = f"{s3_path_parts[0]}/" if len(s3_path_parts) > 1 else ""
+    s3_location = f"s3://{bucket}/{s3_directory}"
```

Comment:
Maybe one to discuss with parking and / or other users down the line but I can envision a world where someone wants to or will organise their csvs differently, or upload a folder of csvs to something like

`s3://bucket/parking/user/{project / collection}/file.csv`